### DISCUSSION TITLE
Probability of Random Events Over Time

### DISCUSSION THREAD

**Discussion by zmanowar** [zmanowar/random-event-analytics#13] [General]
## Random Event Window Probability (Base-Case)
Assuming you stayed logged in between randoms and you remain in an eligible location every 5 minute window.

You have a 0% chance from 0-55 minutes after your last random. Every 5 minutes after that you have a 1/12 chance.
eg. 60 minutes is 1/12, 65 is 2/12, etc.

Specifically the time between `59:59 - 60:01` is 1/12, `64:59 - 65:01` is 2/12. Any times outside of those, eg. `62:00-64:58`, are 0.
_Note: There is probably some leniency in the window, ie. 1 second on either side may not be accurate (it may be more like 10 or 15 seconds), just using 1 second now as an example._

## Ineligible Location
If you are not in an eligible location (discussed in https://github.com/zmanowar/random-event-analytics/discussions/12) during the 60-120 windows, we'll have a bit of a harder time calculating as any missed window could extend your timer by 60-120 minutes.

Presumably you can be in an ineligible location between windows (eg. during `59:01 - 64:59`, `65:01 - 69:59`) will have no affect spawn timer.

### Consequences
If you are in an ineligible location during the window the event has either not spawned yet and the probabilities of the windows have not changed.

Or, if the event was due to spawn [presumably](https://archive.ph/Loos0) all windows/timers are reset and your probability follows the base-case probability model.

## Logged Out
Logging out, hopping worlds, resets the 5 minute window timer. eg. If you log out 3 minutes after logging in you will need to wait a full 5 minutes after logging back in for eligibility

However it's unclear exactly how the global timer and the windows are affected.

_Q: if your random is supposed to spawn in 5 minutes, and you log out and back in on minute 3 -- does that postpone your event 10-70 minutes or does the random occur after the next 5 minute interval? Most likely it postpones it 10-70 minutes._

### Consequences
If you are logged out during the window the event has either not spawned yet and the probabilities of the windows has not changed.
_Or_ the event was due to spawn and your next 5 minute interval has 0% chance, and every window after that has 1/12, 2/12, etc.

Curious about ways to convey a combination of probabilities with the second case or if there's a better way to represent that -- ("metaverse" random event probability branching timelines 🤡).

I'll try to come up with a few scenarios and brush up on my stats.

**Comment by cooperka** [comment]
> Q: if your random is supposed to spawn in 5 minutes, and you log out and back in on minute 3 -- does that postpone your event 10-70 minutes or does the random occur after the next 5 minute interval?

i have played a lot of wintertodt, where games last several minutes with a very short period to bank in between games. i occasionally world-hop to ensure my next 5-minute interval moment occurs during a game break when i'm outside of the event instance (and always avoid banking at that moment). every 5-10 games i'll wait outside for a 5-minute interval and very often get an event -- this makes me think it's the second option: hopping worlds delays by 0-5 minutes (not 10-70).

they key thing seems to be that your are not actively **offline** during a 5-minute interval moment.

**Reply by zmanowar** [reply]
Amazing meta strategy! I'll do some testing as well; that makes sense though, as it would add more complexity to skip an interval (ie. not checking the initial 5-minute) that is probably not appropriate (a la Occam's razor).
